addons/website_event/static/src/js/website_event.editor.js

```diff
@@ -2,11 +2,201 @@ odoo.define('website_event.editor', function (require) {
+        [
+            '/website_event/static/src/xml/website_event_templates.xml'
+        ]
+    ),
+    template: 'website_event.create_event',
+    events: _.extend({}, Dialog.prototype.events, {
+        'change select[name="event_location"]': '_onLocationChanged',
+    }),
+
+    /**
+     * @override
+     * @param {Object} parent
+     * @param {Object} options
+     */
+    init: function (parent, options) {
+        options = _.defaults(options || {}, {
+            title: _t("New Event"),
+            size: 'medium',
+            buttons: [
+                {
+                    text: _t("Create"),
+                    classes: 'btn-primary',
+                    click: this.onCreateClick.bind(this),
+                },
+                {
+                    text: _t("Discard"),
+                    close: true
+                },
+            ]
+        });
+        this._super(parent, options);
+    },
+    start: function () {
+        var self = this;
+        return this._super.apply(this, arguments).then(function () {
+            self.$('div.o_event_form_datetime').each(function () {
+                self._initDateTimePicker($(this));
+            });
+
+            var $input = self.$('#address_id');
+            $input.select2({
+                width: '100%',
+                allowClear: true,
+                formatNoMatches: false,
+                multiple: false,
+                selection_data: false,
+                fill_data: function (query, data) {
+                    var that = this;
+                    var tags = {results: []};
+                    _.each(data, function (obj) {
+                        var display_text = obj.display_name + ': ' + obj.street + ', ' + (obj.zip ? (obj.zip + ', ') : '') + obj.city + ', ' + obj.country_id[1];
+                        if (that.matcher(query.term, display_text)) {
+                            tags.results.push({id: obj.id, text: display_text});
+                        }
+                    });
+                    query.callback(tags);
+                },
+                query: function (query) {
+                    var that = this;
+                    // fetch data only once and store it
+                    if (!this.selection_data) {
+                        self._rpc({
+                            model: 'res.partner',
+                            method: 'search_read',
+                            domain: [['company_id', '=', false]],
+                            fields: ['display_name', 'street', 'zip', 'city', 'country_id'],
+                        }).then(function (data) {
+                            that.fill_data(query, data);
+                            that.selection_data = data;
+                        });
+                    } else {
+                        this.fill_data(query, this.selection_data);
+                    }
+                }
+            });
+        });
+    },
+    _initDateTimePicker: function ($dateGroup) {
```

Comment:
Don't we have anything available at hand to avoid doing such low-level code ? Like widgets, as normally you have widgets available in qweb-xml ?

Reply:
Maybe I missed something but I couldn't find another datetimepicker that would allow us to restrict date choice while also displaying the time at the right format.

Our dear DBE had to use the same thing in survey.